### PR TITLE
Set up documentation

### PR DESCRIPTION
**Pull request by jb-mackay**
Sets up Documenter.jl infrastructure and adds documentation for linear solvers and preconditioners.

Removes extra PCInitialize calls in specific solver implementations.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,6 @@ docs/site/
 # committed for packages, but should be committed for applications that require a static
 # environment.
 Manifest.toml
+
+#Temporary
+*.DS_Store
```
Comment (simonbyrne) [inline .gitignore:27]: this is an OS-specific file. It should probably go in your global `core.excludesFile`, which by default is `~/.config/git/ignore`. See https://git-scm.com/docs/gitignore

Comment (simonbyrne) [reply .gitignore:27]: and then you can delete the `git rm docs/.DS_Store`

File: docs/make.jl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    ]
+)
+
+# Documenter can also automatically deploy documentation to gh-pages.
+# See "Hosting Documentation" and deploydocs() in the Documenter manual
+# for more information.
+# deploydocs(
+#     repo = "github.com/CliMA/Solvent.jl.git",
+#     push_preview = true,
+# )
```
Comment (simonbyrne) [inline docs/make.jl:28]: uncomment this.

**Review by thomasgibson — CHANGES_REQUESTED** [review]
Please remove `docs/.DS_Store` then this should be good to go!

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**